templates/crossplane/pkg/controller.go.tpl

```diff
@@ -73,13 +64,18 @@ func (e *external) Observe(ctx context.Context, mg cpresource.Managed) (managed.
 
 {{- if .CRD.Ops.ReadOne }}
 	input := Generate{{ .CRD.Ops.ReadOne.InputRef.Shape.ShapeName }}(cr)
+	if err := e.preObserve(ctx, cr, input); err != nil {
+  	return managed.ExternalObservation{}, errors.Wrap(err, "pre-observe failed")
+  }
```

Comment:
nit: indentation is off above andf throughout the added lines here in this template. For `.tpl` files, probably need to `s/\ \ \ \ /\t/gI`